pkg/controller/daemon/daemoncontroller.go

```diff
@@ -247,23 +256,33 @@ func (dsc *DaemonSetsController) Run(workers int, stopCh <-chan struct{}) {
 
 	<-stopCh
 	glog.Infof("Shutting down Daemon Set Controller")
-	dsc.queue.ShutDown()
 }
 
 func (dsc *DaemonSetsController) runWorker() {
-	for {
-		dsKey, quit := dsc.queue.Get()
-		if quit {
-			continue
-		}
-		err := dsc.syncHandler(dsKey.(string))
-		if err != nil {
-			glog.Errorf("Error syncing daemon set with key %s: %v", dsKey.(string), err)
-		}
-		dsc.queue.Done(dsKey)
+	for dsc.processWork() {
 	}
 }
 
+// processWork deals with one key off the queue.  It returns false when it's time to quit
+func (dsc *DaemonSetsController) processWork() bool {
```

Comment:
maybe make the "work for one key" property explicit in the function name somehow